ynr/apps/sopn_parsing/helpers/extract_pages.py

```diff
@@ -26,80 +27,47 @@ def extract_pages_for_ballot(ballot, manual_upload=False):
-        # the new pdf file.
-        new_pdf_official_document = OfficialDocument(
-            ballot=ballot,
-            source_url=ballot.sopn.source_url,
-            document_type=OfficialDocument.NOMINATION_PAPER,
-        )
+        html_file_path = ballot.sopn.uploaded_file.path
 
-        # save the uploaded file data to the new pdf offical doc
-        # TO DO: How can I get the content file object such as...
-        # file_object = ContentFile(content=file_response.content)
+        with tempfile.NamedTemporaryFile(suffix=".pdf") as temp_file:
+            # convert the doc and save to a temp file
+            pypandoc.convert_file(
+                html_file_path, to="pdf", outputfile=temp_file.name
+            )
 
-        new_pdf_official_document.uploaded_file.save(
-            name=pdf_file_name,
-            # content=file_object
-        )
+            # save the pdf file to the sopn document
+            pdf_file_name = ballot.sopn.uploaded_file.name.replace(
+                ".html", ".pdf"
+            )
 
-        # get the pdf file
-        document = new_pdf_official_document
-
-        html_document.delete()
+            ballot.sopn.uploaded_file.save(
+                name=pdf_file_name, content=temp_file, save=True
+            )
+            ballot.sopn.save()
 
     elif ballot.sopn.uploaded_file.path.endswith(".docx"):
         print(
             f"Warning! docx file types are not supported; trying to convert {ballot.sopn.uploaded_file.path} to pdf"
         )
 
         docx_file_path = ballot.sopn.uploaded_file.path
-        docx_file_name = ballot.sopn.uploaded_file.name
-        pdf_file_path = docx_file_path + ".pdf"
 
-        pypandoc.convert_file(
-            docx_file_path, to="docx", outputfile=docx_file_path + ".pdf"
-        )
+        with tempfile.NamedTemporaryFile(suffix=".pdf") as temp_file:
+            pypandoc.convert_file(
+                docx_file_path, to="pdf", outputfile=temp_file.name
+            )
 
-        pdf_file_path.replace(".docx", "")
-        pdf_file_name = docx_file_name.replace(".docx", "")
+            pdf_file_name = ballot.sopn.uploaded_file.name.replace(
+                ".docx", ".pdf"
+            )
 
-        # get the docx file
-        docx_document = OfficialDocument.objects.filter(
-            source_url=ballot.sopn.source_url
-        ).order_by("-uploaded_file")[0]
-
-        # Create a new OfficialDocument object with the same source_url but with
-        # the new pdf file.
-        new_pdf_official_document = OfficialDocument(
-            ballot=ballot,
-            source_url=ballot.sopn.source_url,
-            document_type=OfficialDocument.NOMINATION_PAPER,
-        )
-
-        # get the pdf file
-        document = new_pdf_official_document
-
-        docx_document.delete()
+            ballot.sopn.uploaded_file.save(
+                name=pdf_file_name, content=temp_file, save=True
+            )
+            ballot.sopn.save()
```

Comment:
I think `pdf_file_name` needs revision; This save is creating a nested file folder such as 

` 'uploaded_file': 'official_documents/local.harrow.pinner-south.by.2021-10-14/official_documents/local.harrow.pinner-south.by.2021-10-14/Democracy_Club_Candidates.pdf'`

Reply:
This is because `ballot.sopn.uploaded_file.name` actually includes the full path. Try changing to:
```
pdf_file_name = ballot.sopn.uploaded_file.name.split("/")[-1].replace(".docx", ".pdf")
```